beacon-chain/sync/fork_watcher.go

```diff
@@ -42,67 +41,46 @@ func (s *Service) forkWatcher() {
 
 // registerForUpcomingFork registers appropriate gossip and RPC topic if there is a fork in the next epoch.
 func (s *Service) registerForUpcomingFork(currentEpoch primitives.Epoch) error {
-	// Get the genesis validators root.
-	genesisValidatorsRoot := s.cfg.clock.GenesisValidatorsRoot()
-
+	nextEntry := params.GetNetworkScheduleEntry(currentEpoch + 1)
 	// Check if there is a fork in the next epoch.
-	isForkNextEpoch, err := forks.IsForkNextEpoch(s.cfg.clock.GenesisTime(), genesisValidatorsRoot[:])
-	if err != nil {
-		return errors.Wrap(err, "Could not retrieve next fork epoch")
-	}
-
-	// Exit early if there is no fork in the next epoch.
-	if !isForkNextEpoch {
+	if nextEntry.ForkDigest == s.registeredNetworkEntry.ForkDigest {
```

Comment:
Can we use `params.DigestChangesAfter` here? That would eliminate the need for `s.registeredNetworkEntry`